tool_watershed/watershed_analysis_networkx.py

```diff
@@ -235,10 +242,18 @@ def cells_as_multipolygon(self, cell_ids: set):
 
         WARNING! May result in memory error if somehwat large set of cells is given as input
         """
-        upstream_cells = self._gr.cells.filter(id__in=list(cell_ids))
+        # upstream_cells = self._gr.cells.filter(id__in=list(cell_ids))
         cell_drv = ogr.GetDriverByName("MEMORY")
         cell_ds = cell_drv.CreateDataSource("")
-        threedigrid_to_ogr(threedigrid_src=upstream_cells, tgt_ds=cell_ds, attributes={}, attr_data_types={})
+        result = self.model.get_result(self.result_id)
+        gridadmin_gpkg = result.parent().path.with_suffix('.gpkg')
+        threedigrid_to_ogr(
+            tgt_ds=cell_ds,
+            layer_name="cell",
```

Comment:
Passing just ``layer_name="cell"`` to ``threedigrid_to_ogr`` instead of just the ``upstream_cells`` will lead to a very different algorithm output. It will now make a multipolygon of all cells in the threedimodel, instead of just the cells upstream of the point that was clicked by the user. 

This makes me realize that ``threedigrid_to_ogr`` needs another argument ``ids: Sequence[int]``, so that you can request a subset of nodes/cells/flowlines. this is relevant for all the tools that use it (watershed tool, result aggregation, and cross-sectional discharge)

Reply:
Resolved